content/lib/2021-04-02-operating-system-interface.md

```diff
@@ -0,0 +1,225 @@
+    - Official [Open Container Image Spec](https://github.com/opencontainers/image-spec)
+
+CRI is a implementation of OCI.
+The main difference is scope, in which case OCI also defines the [open container image spec](https://github.com/opencontainers/image-spec).
+
+
+- OCI and CRI define the container runtime API surface/shape for interfacing with container runtime "engines" from `userland`. 
+    - OCI also defines the API surface/shape and interface for packaging container images
+- Container runtimes are tightly coupled with Linux features, and therefor are tightly coupled with the operating system. 
+- Container runtimes manage permissions, storage, and networking in different ways. 
+- Container runtimes are tightly coupled with `storage` and `networking`.
+- Container runtimes do not define *how* they are installed or managed. 
+
+# The Void
+
+There are several `userland` components that are not considered from a management perspective. 
+
+Below are demonstrable questions that technically have answers.
+I hope to challenge the existing answers to these questions with DOSi.
+
+### Demonstrable Questions
+
+- How is my container network defined?
+- How are my storage drivers defined?
+- How is my container runtime defined?
+    - Which container runtime will I use?
+    - How will I upgrade my container runtime?
+- Which clients have access to my container runtime?
+- How do I mutate my network configuration in `userland`?
+- How do I mutate my network configuration in `kubernetesland`?
+- How do I mutate kernel configuration? 
+    - eBPF
+    - Kernel modules
+    - IPTables
+    - Seccomp filters and Seccomp policy
+    - Namespace sharing
+    - Linux capabilities
+- How can I read kernel runtime values?
+
+# Concrete use cases
+
+Below are some high level examples of concrete use cases that would directly benefit from having a standardized distributed operating system level interface (DOSi).
+
+### Patching openssh 
+
+In 2020 [CVE-2020-15778](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-15778) was announced which allows command injection via `scp` using the `scp.c` function `toremote()` in the `OpenSSH` library.
+
+What is the existing story to update the `userland` libraries? 
+How would `kubernetesland` be impacted by this update?
+
+### Managing Access Control and Security Software
+
+Below find concrete examples of security software that will span `kubernetesland`, `userland`, and `kernelland`. 
+Currently, there is no clearly (or safely) defined way to manage or mutate these concerns from a single control plane.
+
+ - Seccomp policy
+ - Pod Security Policy
+ - Kubernetes Admissions Control Policy
+ - Falco runtime security
+ - AppArmor Policy
+
+### Package Management 
+
+Any `userland` level package is traditionally managed on a per instance basis and is managed via `userland`.
+Furthermore, the kernel, operating system, and operating system dependencies also need to be managed. 
```

Comment:
These are great examples of this idea of "push as much as we can above the node" that I personally think is important. Nodes are things we should more or less forget about and we should look at the cluster as a single machine.